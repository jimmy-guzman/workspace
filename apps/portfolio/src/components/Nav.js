import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'

const Nav = ({ isEnglish }) => (
  <nav className='main-nav'>
    <NavLink to='/about'>{isEnglish ? 'About' : 'Sobre Mi'}</NavLink>
    <NavLink to='/projects'>{isEnglish ? 'Projects' : 'Proyectos'}</NavLink>
    <NavLink to='/contact'>{isEnglish ? 'Contact' : 'Contacto'}</NavLink>
  </nav>
)

Nav.propTypes = {
  isEnglish: PropTypes.bool.isRequired
}

export default Nav
