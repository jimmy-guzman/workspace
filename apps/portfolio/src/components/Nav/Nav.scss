@function em($target, $context: 18px) {
  @return ($target/$context) * 1em;
}

.main-nav {
  font-family: 'Oswald', Impact, 'Franklin Gothic Bold', sans-serif;
  background: #292b2f;
  width: 100%;
  height: em(70px);
  display: flex;
  flex-flow: row;
  justify-content: space-around;
  align-items: center;
  a {
    font-size: em(24px);
    text-decoration: none;
    color: #d7d5d9;
    fill: #d7d5d9;
    &:hover,
    &.active {
      color: #40af93;
      fill: #40af93;
    }
  }
}
