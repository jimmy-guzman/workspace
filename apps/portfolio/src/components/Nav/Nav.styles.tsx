import styled from 'styled-components'

import { em } from '@portfolio/utils'

const Nav = styled.nav`
  background: ${({ theme }) => theme.colors.tertiary};
  width: 100%;
  height: ${em('74px')};
  display: flex;
  flex-flow: row;
  justify-content: space-around;
  align-items: center;
`

export const Styled = { Nav }
