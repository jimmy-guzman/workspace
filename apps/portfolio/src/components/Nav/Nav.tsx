import React from 'react'
import { NavLink } from 'react-router-dom'

export interface NavProps {
  isEnglish?: boolean
}

export const Nav = ({ isEnglish = true }: NavProps) => (
  <nav className='main-nav'>
    <NavLink to='/about'>{isEnglish ? 'About' : 'Sobre Mi'}</NavLink>
    <NavLink to='/projects'>{isEnglish ? 'Projects' : 'Proyectos'}</NavLink>
    <NavLink to='/contact'>{isEnglish ? 'Contact' : 'Contacto'}</NavLink>
  </nav>
)
