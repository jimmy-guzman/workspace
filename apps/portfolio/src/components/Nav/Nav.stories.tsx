import React from 'react'
import { Nav, defaultLinks } from './Nav'
import { withMockRouter, withThemeProvider } from '../../utils'

export default {
  title: 'portfolio/components/Nav',
  component: Nav,
  decorators: [withMockRouter, withThemeProvider],
}

export const normal = () => <Nav links={defaultLinks} />
export const spanish = () => <Nav links={defaultLinks} isEnglish={false} />
