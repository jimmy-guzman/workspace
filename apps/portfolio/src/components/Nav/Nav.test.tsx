import React from 'react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import { render } from '@testing-library/react'

import { Nav, NavProps } from './Nav'

const setupNav = (props?: NavProps) => {
  const history = createMemoryHistory()

  return render(
    <Router history={history}>
      <Nav {...props} />
    </Router>
  )
}

describe('<Nav />', () => {
  it('should render default snapshot', () => {
    const { container } = setupNav()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render snapshot with not isEnglish', () => {
    const { container } = setupNav({ isEnglish: false })

    expect(container.firstChild).toMatchSnapshot()
  })
})
