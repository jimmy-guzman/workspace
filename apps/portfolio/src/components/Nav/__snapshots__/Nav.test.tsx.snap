// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render default snapshot 1`] = `
.c0 {
  background: #292b2f;
  width: 100%;
  height: 4.11em;
  display: flex;
  flex-flow: row;
  justify-content: space-around;
  align-items: center;
}

.c1 {
  font-family: 'Oswald',Impact,'Franklin Gothic Bold',sans-serif;
  text-decoration: none;
  color: #d7d5d9;
  fill: #d7d5d9;
  font-size: 1.33em;
}

.c1:hover {
  color: #40af93;
  fill: #40af93;
  cursor: pointer;
}

.c1:active,
.c1.active {
  color: #40af93;
  fill: #40af93;
}

<nav
  class="c0"
>
  <a
    class="c1"
    color="primary"
    href="/about"
  >
    About
  </a>
  <a
    class="c1"
    color="primary"
    href="/projects"
  >
    Projects
  </a>
  <a
    class="c1"
    color="primary"
    href="/contact"
  >
    Contact
  </a>
</nav>
`;

exports[`<Nav /> should render snapshot with not isEnglish 1`] = `
.c0 {
  background: #292b2f;
  width: 100%;
  height: 4.11em;
  display: flex;
  flex-flow: row;
  justify-content: space-around;
  align-items: center;
}

.c1 {
  font-family: 'Oswald',Impact,'Franklin Gothic Bold',sans-serif;
  text-decoration: none;
  color: #d7d5d9;
  fill: #d7d5d9;
  font-size: 1.33em;
}

.c1:hover {
  color: #40af93;
  fill: #40af93;
  cursor: pointer;
}

.c1:active,
.c1.active {
  color: #40af93;
  fill: #40af93;
}

<nav
  class="c0"
>
  <a
    class="c1"
    color="primary"
    href="/about"
  >
    Sobre Mi
  </a>
  <a
    class="c1"
    color="primary"
    href="/projects"
  >
    Proyectos
  </a>
  <a
    class="c1"
    color="primary"
    href="/contact"
  >
    Contacto
  </a>
</nav>
`;
